src/west/app/project.py

```diff
@@ -602,6 +602,118 @@ def _dump(self, args, to_dump):
         else:
             sys.stdout.write(to_dump)
 
+class Compare(_ProjectCommand):
+    def __init__(self):
+        super().__init__(
+            'compare',
+            "compare project status against the manifest",
+            textwrap.dedent('''\
+            Compare each project's working tree state against the results
+            of the most recent successful "west update" command.
+
+            This command prints output for a project if (and only if)
+            at least one of the following is true:
+
+            1. its checked out commit is different than the commit checked
+               out by the most recent successful "west update"
+            2. its working tree is not clean (i.e. there are local uncommitted
+               changes)
+            3. it has a local checked out branch
+
+            The command prints output for the manifest repository if it
+            has nonempty status.
+
+            The output is meant to be human-readable, and may change. It is
+            not a stable interface to write scripts against.''')
```

Comment:
should we add a line that states that git >=2.22 is required for this command, so that it's also mentioned in the help text. After all, help text is a form of documentation for `west`.
I do know we also have docs here: https://docs.zephyrproject.org/latest/develop/west/index.html, but `west help <cmd>` is faster 😉